subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -74,14 +99,14 @@ static void handle_get(struct bt_mesh_model *mod, struct bt_mesh_msg_ctx *ctx,
 
 static void scene_set(struct bt_mesh_scene_srv *srv, uint16_t scene)
 {
-	if (srv->curr == scene) {
+	if (srv->next == scene) {
 		return;
 	}
 
-	srv->curr = scene;
+	srv->next = scene;
 	if (scene) {
```

Comment:
I see you have changed these to `if (scene != BT_MESH_SCENE_NONE)` everywhere, probably should do that here as well? Only nitpick, otherwise this looks OK now.

Reply:
Fixed